lib/omniauth-ethereum.rb

```diff
@@ -13,22 +14,54 @@ class Ethereum
 
       # the `eth_address` will be the _fake_ unique identifier for the Ethereum strategy
       option :uid_field, :eth_address
+      option :fields, [:eth_message, :eth_address, :eth_signature]
+      option :uid_field, :eth_address
 
-      # the omniauth request phase
       def request_phase
-
-        # helper omniauth form to gather required data
-        form = OmniAuth::Form.new :title => "Ethereum Authentication", :url => callback_path
+        form = OmniAuth::Form.new :title => 'Ethereum Authentication', :url => callback_path
         options.fields.each do |field|
 
           # these fields are read-only and will be filled by javascript in the process
-          form.text_field field.to_s.capitalize.gsub("_", " "), field.to_s, readonly: true, class: field.to_s
+          # we were using readonly here before, why?
```

Comment:
see https://github.com/q9f/ethereum-on-rails/blob/main/app/views/sessions/new.html.erb#L14-L16

the user does not need to fill in anything into these forms. by making it read-only, I ensure everything is handled by the JS logic.

Reply:
Updated to make these readonly. The form builder is pretty basic, so had to drop down to html.